### PR TITLE
New, prime, tables

### PR DESCRIPTION
**Pull request by rpiaggio**
- Migrate all tables to new version of `react-tabe` + `react-virtual`, rendering with `react-prime` styles.
- Column selector was extracted into a component in `lucuma-ui`.

A word about `SpectroscopyModesTable`: Behavior was modified slightly. Visible rows are no longer prioritized when querying ITC. This is because our way of prioritizing was cancelling current queries and querying again in different order upon scroll. This could cause a lot of cancelled requests if the user continually scrolled. We can reimplement priorization with a different approach, ideally without cancelling requests already in flight.

### CODE REVIEW

File: common/src/main/scala/explore/Icons.scala [hunk 1]
```diff
@@ -130,18 +130,6 @@ object Icons {
   @JSImport("@fortawesome/pro-solid-svg-icons", "faCheck")
   val faCheck: FAIcon = js.native
 
-  @js.native
-  @JSImport("@fortawesome/pro-solid-svg-icons", "faSort")
```
Comment (cquiroz) [inline common/src/main/scala/explore/Icons.scala:134]: Are these provided by prime?

Comment (rpiaggio) [reply common/src/main/scala/explore/Icons.scala:134]: We are still using the same FA icons, but they are declared in `lucuma-ui` now.

File: common/src/main/webapp/sass/explore.scss [hunk 2]
```diff
@@ -222,13 +224,11 @@ $search-preview-margin: 5px;
...
   align-items: center;
-  background: repeating-linear-gradient(
-    -45deg,
-    black,
-    black 10px,
-    var(--site-border-color) 10px,
-    var(--site-border-color) 11px
-  );
+  background: repeating-linear-gradient(-45deg,
+      black,
```
Comment (cquiroz) [inline common/src/main/webapp/sass/explore.scss:228]: I guess this should be a theme variable?

**Review by cquiroz — APPROVED** [review]
LGTM. It is a bit hard to check everything but so far it seems fine
Thanks a lot for doing this

File: common/src/main/webapp/sass/explore.scss (hunk 2 above)
Comment (rpiaggio) [reply common/src/main/webapp/sass/explore.scss:228]: Agreed. This was not introduced in this PR though. It was autoformatted by prettier.

**Review by toddburnside — APPROVED** [review]
LGTM

**Pull request merged**